diff --git a/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-config.js b/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-config.js
index 2ac0986..246348c 100644
--- a/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-config.js
+++ b/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-config.js
@@ -5,7 +5,7 @@ const devkit_1 = require("@nx/devkit");
 const path_1 = require("path");
 function normalizeStyleDictionaryConfig(config, options, context) {
     var _a;
-    const { root: projectRoot } = context.workspace.projects[context.projectName];
+    const { root: projectRoot } = context.projectsConfigurations.projects[context.projectName];
     const normalized = Object.assign(Object.assign({}, config), { source: config.source.map((src) => path_1.posix.resolve(projectRoot, src)), include: (_a = config === null || config === void 0 ? void 0 : config.include) === null || _a === void 0 ? void 0 : _a.map((include) => path_1.posix.resolve(projectRoot, include)), platforms: Object.entries(config.platforms).reduce((accum, [name, platform]) => (Object.assign(Object.assign({}, accum), { [name]: Object.assign(Object.assign({}, platform), { buildPath: (0, devkit_1.joinPathFragments)(options.outputPath, platform.buildPath) }) })), {}) });
     return normalized;
 }
diff --git a/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-options.js b/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-options.js
index 08106c2..809142c 100644
--- a/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-options.js
+++ b/node_modules/@nxkit/style-dictionary/src/executors/build/lib/normalize-options.js
@@ -4,7 +4,7 @@ exports.normalizePluginPath = exports.normalizeOptions = void 0;
 const path_1 = require("path");
 function normalizeOptions(options, context) {
     var _a;
-    const projectConfig = context.workspace.projects[context.projectName];
+    const projectConfig = context.projectsConfigurations.projects[context.projectName];
     const { sourceRoot, root: projectRoot } = projectConfig;
     const { root } = context;
     return Object.assign(Object.assign({}, options), { root,
